numpy/distutils/exec_command.py

```diff
@@ -249,14 +249,28 @@ def _exec_command(command, use_shell=None, use_tee = None, **env):
         proc = subprocess.Popen(command, shell=use_shell, env=env,
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.STDOUT,
-                                universal_newlines=True)
+                                universal_newlines=False)
+        # universal_newlines is set to False so that communicate()
+        # will return bytes. We need to decode the output ourselves
+        # so that Python will not raise a UnicodeDecodeError when
+        # it encounters an invalid character; rather, we simply replace it
     except EnvironmentError:
         # Return 127, as os.spawn*() and /bin/sh do
         return 127, ''
+
     text, err = proc.communicate()
+    text = text.decode(sys.getdefaultencoding(), errors='replace')
+
+    text = text.replace('\r\n', '\n')
     # Another historical oddity
     if text[-1:] == '\n':
         text = text[:-1]
+
+    # stdio uses bytes in python 2, so to avoid issues, we simply
+    # remove all non-ascii characters
+    if sys.version_info < (3, 0):
+        text = text.encode('ascii', errors='replace')
```

Comment:
To me it seems that this encoding should be happening where `print` is called, not in the function itself